spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsMessageConversionDelegate.java

```diff
@@ -164,22 +163,22 @@ public void init(ProcessorContext context) {
 
 			@Override
 			public void process(Object o, Object o2) {
-				if (kstreamBindingInformationCatalogue.isDlqEnabled(bindingTarget)) {
-					String destination = context.topic();
-					if (o2 instanceof Message) {
-						Message message = (Message) o2;
-						sendToDlqAndContinue.sendToDlq(destination, (byte[]) o, (byte[]) message.getPayload(), context.partition());
-					}
-					else {
-						sendToDlqAndContinue.sendToDlq(destination, (byte[]) o, (byte[]) o2, context.partition());
+				//Only continue if the record was not a tombstone.
+				if (o2 != null) {
```

Comment:
Why can't we send to DLQ a null? Sounds like a valid use-case for Kafka.
Or at least some representation for null, or simple String mention that null is sent to DQL...

I will accept any explanation, but still would like to raise a concern for further thoughts.

Reply:
The record is a tombstone and I don't think we should send that to the DLQ as it is. I like the suggestion of a string representation of the tombstone record. Haven't seen such use cases or feature requests before, but worth looking into as a future enhancement.

Reply:
Right. That's fine to move it into the future. Meanwhile let's consider at least some INFO as an `else` variant. 
Otherwise there might be some question where has that gone...

Reply:
The `else` case is addressed where the tombstone record is processed. This one is specifically for handling deserialization errors and since we already log the message as part of the initial processing, I don't think we need to log it again.